### DISCUSSION TITLE
How to make a post request with /server/api/ routes?

### DISCUSSION THREAD

**Discussion by christopher4lis** [nuxt/nuxt#18326] [Questions] [answered]
I see that the `server/api/` routes directory is great for making `GET` requests, but is there a way to process other request types such as `POST`, `PUT`, and `DELETE`?

**Comment by HenrijsS** [comment, marked as answer]
The API directory uses Nitro, which uses H3, which uses Node HTTP.

You have the default Node HTTP object properties which you can check. Check here: https://nodejs.org/api/http.html#http_http_request_options_callback

Now for the request. Just check the method:
```js
export default (req, res) => {
	if (req.method == 'POST') {
		return 'Post method';
	} else if (req.method == 'GET') {
		return 'Get method';
	}
};
```

The `req` object has a property called `method` which corresponds to the request.
